# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webcodecs/AudioData.cpp
# Commit: f86f5e52c4bb
# Full Hash: f86f5e52c4bbe09185a212e948f6a0bd433b777c
# Author: Paul Adenot <paul@paul.cx>
# Date: 2024-03-07 09:55:57
# Regressor Bug: 1858958
# File Overlap Count: 1
# Description:
#   Bug 1858958 - Test more conversions, clearly mark planar->interleaved conversion as outside of spec. r=chunmin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D192339
# ==============================================================================

diff -r 8814c82ce54b -r f86f5e52c4bb dom/media/webcodecs/AudioData.cpp
--- a/dom/media/webcodecs/AudioData.cpp	Wed Mar 06 14:00:16 2024 +0000
+++ b/dom/media/webcodecs/AudioData.cpp	Wed Mar 06 14:00:16 2024 +0000
@@ -443,6 +443,7 @@
   }
 
   if (!IsInterleaved(aSourceFormat) && IsInterleaved(aCopyToSpec.mFormat)) {
+    MOZ_CRASH("This should never be hit -- current spec doesn't support it");
     // Planar to interleaved -- copy of all channels of the source into the
     // destination buffer.
     MOZ_ASSERT(aCopyToSpec.mPlaneIndex == 0);